src/v/kafka/server/group.cc

```diff
@@ -3410,7 +3325,12 @@ group::get_expired_offsets(std::chrono::seconds retention_period) {
 
 bool group::has_offsets() const {
     return !_offsets.empty() || !_pending_offset_commits.empty()
-           || !_tx_data.empty();
+           || std::any_of(
+             _producers.begin(),
+             _producers.end(),
+             [](const producers_map::value_type& p) {
+                 return p.second.transaction != nullptr;
```

Comment:
should we also check that `transaction.offsets` is not empty?

Reply:
previously if the transaction was present we returned `true` from `has_offsets` this is in align with previous logic.